File: android/src/main/java/com/video/hybrids/HybridVideoPlayerFactory.kt
package com.margelo.nitro.video

import androidx.annotation.OptIn
import androidx.media3.common.util.UnstableApi
import com.facebook.proguard.annotations.DoNotStrip

@DoNotStrip
class HybridVideoPlayerFactory(): HybridVideoPlayerFactorySpec() {
  @OptIn(UnstableApi::class)
  override fun createPlayer(source: HybridVideoPlayerSourceSpec): HybridVideoPlayerSpec {
    return HybridVideoPlayer(source as HybridVideoPlayerSource)
  }

  override val memorySize: Long
    get() = 0
}

File: packages/react-native-video/ios/hybrids/VideoPlayer/HybridVideoPlayerFactory.swift
//
//  HybridVideoPlayerFactory.swift
//  ReactNativeVideo
//
//  Created by Krzysztof Moch on 09/10/2024.
//

import Foundation
import NitroModules

class HybridVideoPlayerFactory: HybridVideoPlayerFactorySpec {
  func createPlayer(source: HybridVideoPlayerSourceSpec) throws -> HybridVideoPlayerSpec {
    return try HybridVideoPlayer(source: source)
  }
}
